File: ueueo-framework/ueueo-localization/src/main/java/Abp/Localization/LocalizationResource.cs
using System;
using System.Collections.Generic;
using System.Linq;
using JetBrains.Annotations;

namespace Volo.Abp.Localization;

public class LocalizationResource
{
    [NotNull]
    public Type ResourceType;//  { get; }

    [NotNull]
    public String ResourceName => LocalizationResourceNameAttribute.GetName(ResourceType);

    [CanBeNull]
    public String DefaultCultureName;// { get; set; }

    [NotNull]
    public LocalizationResourceContributorList Contributors;//  { get; }

    [NotNull]
    public List<Type> BaseResourceTypes;//  { get; }

    public LocalizationResource(
        @Nonnull Type resourceType,
        @Nullable String defaultCultureName = null,
        @Nullable ILocalizationResourceContributor initialContributor = null)
    {
        ResourceType = Objects.requireNonNull(resourceType, nameof(resourceType));
        DefaultCultureName = defaultCultureName;

        BaseResourceTypes = new List<Type>();
        Contributors = new LocalizationResourceContributorList();

        if (initialContributor != null)
        {
            Contributors.Add(initialContributor);
        }

        AddBaseResourceTypes();
    }

    protected   void AddBaseResourceTypes()
    {
        var descriptors = ResourceType
            .GetCustomAttributes(true)
            .OfType<IInheritedResourceTypesProvider>();

        for (var descriptor in descriptors)
        {
            for (var baseResourceType in descriptor.GetInheritedResourceTypes())
            {
                BaseResourceTypes.AddIfNotContains(baseResourceType);
            }
        }
    }
}

File: ueueo-framework/ueueo-localization/src/main/java/com/ueueo/localization/LocalizationResource.java
package com.ueueo.localization;

import lombok.Data;
import org.apache.commons.lang3.ArrayUtils;
import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

@Data
public class LocalizationResource {
    @NonNull
    private Class<?> resourceType;

    @Nullable
    private String defaultCultureName;

    @NonNull
    private LocalizationResourceContributorList contributors;

    @NonNull
    private Set<Class<?>> baseResourceTypes;

    public String getResourceName() {
        LocalizationResourceNameAttribute attribute = resourceType.getAnnotation(LocalizationResourceNameAttribute.class);
        if (attribute != null) {
            return attribute.name();
        } else {
            return null;
        }
    }

    public LocalizationResource(
            @NonNull Class<?> resourceType,
            @Nullable String defaultCultureName,
            @Nullable ILocalizationResourceContributor initialContributor) {
        this.resourceType = Objects.requireNonNull(resourceType);
        this.defaultCultureName = defaultCultureName;

        baseResourceTypes = new HashSet<>();
        contributors = new LocalizationResourceContributorList();

        if (initialContributor != null) {
            contributors.add(initialContributor);
        }
        addBaseResourceTypes();
    }

    protected void addBaseResourceTypes() {
        InheritResourceAttribute attribute = resourceType.getAnnotation(InheritResourceAttribute.class);
        if (attribute != null && ArrayUtils.isNotEmpty(attribute.resourceTypes())) {
            baseResourceTypes.addAll(Arrays.asList(attribute.resourceTypes()));
        }
    }
}
